Source/ActionController.swift

```diff
@@ -361,23 +427,30 @@ open class ActionController<ActionViewType: UICollectionViewCell, ActionDataType
                 return reusableview!
             } else {
                 let reusableview = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: ReusableViewIds.SectionHeader.rawValue, for: indexPath) as? SectionHeaderViewType
-                onConfigureSectionHeader?(reusableview!,  sectionForIndex(actionSectionIndexFor((indexPath as NSIndexPath).section))!.data!)
+                onConfigureSectionHeader?(reusableview!, sectionForIndex(actionSectionIndexFor((indexPath as NSIndexPath).section))!.data!)
                 return reusableview!
             }
         }
-        
         fatalError()
     }
-    
+
     open func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+
         let action = actionForIndexPath(actionIndexPathFor(indexPath))
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ReusableViewIds.Cell.rawValue, for: indexPath) as? ActionViewType
-        self.onConfigureCellForAction?(cell!, action!, indexPath)
-        return cell!
+        if let action = action {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ReusableViewIds.Cell.rawValue, for: indexPath) as! ActionViewType
+            self.onConfigureCellForAction?(cell, action, indexPath)
+            return cell
+        } else {
+            // check if cancel should be added
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ReusableViewIds.Cancel.rawValue, for: indexPath) as! CancelViewType
```

Comment:
I'm not sure about having the cancel view as another cell. I think it will make things a bit harder and error prune. Is there a reason to having it as a cell instead of the footer of the last section?